minari/data_collector/data_collector.py

```diff
@@ -230,7 +230,12 @@ def step(
         # or truncation. This may happen if the step_data_callback truncates or terminates the episode under
         # certain conditions.
         if self._new_episode and not self._reset_called:
-            self._buffer[-1]["observations"] = [self._previous_eps_final_obs]
+            if isinstance(self._previous_eps_final_obs, dict):
```

Comment:
Fix for dictionary observation spaces

Reply:
If possible, I think it would be good to add a test for this behavior.